src/plugins/tracking/util/clickState.js

```diff
@@ -0,0 +1,56 @@
+import JSCookie from 'js-cookie';
+
+/*
+ * This module provides utilities for managing click tracking data in Redux
+ * state an saving that data to a browser cookie
+ */
+
+export const COOKIE_NAME = 'click-track'; // must remain in sync with Meetup Classic implementation
+
+const BrowserCookies = JSCookie.withConverter({
+	read: (value, name) => value,
+	write: (value, name) =>
+		encodeURIComponent(value).replace(
+			/[!'()*]/g,
+			c => `%${c.charCodeAt(0).toString(16)}`
+		),
+});
+
+export const setClickCookie = clickTracking => {
+	const domain = window.location.host.replace(/[^.]+/, ''); // strip leading subdomain, e.g. www or beta2
+	const cookieVal = JSON.stringify(clickTracking);
+	BrowserCookies.set(COOKIE_NAME, cookieVal, { domain });
+};
+
+export const CLICK_TRACK_ACTION = 'CLICK_TRACK';
+export const CLICK_TRACK_CLEAR_ACTION = 'CLICK_TRACK_CLEAR';
+
+export const actions = {
```

Comment:
I'm not really sure what the 'best' solution is for this since this module lives inside the 'click'-related code, so it's implicit that the `actions` are '`clickActions` and the `reducer` is a `clickReducer` - I could rename but it's a little redundant from the perspective of the module. Also, it makes sense if the whole module is imported and the named exports get referenced from a top-level object, e.g.

```js
import * as clickState from '.../clickState';

clickState.actions.click();
clickData: clickState.reducer(state);
```

Since the name of the typically ends up being where it stores its corresponding `state` data, the consumer app will probably always want to alias it in some way just so that it has control over where the data appears, but the name of the `actions` export could be made more unique. I'm ambivalent.